### ISSUE TITLE
Fill the output of af_create_indexers with af_span by default

### ISSUE THREAD

**Issue by ranjanan** [arrayfire/arrayfire#1425]
I wrote the following test program in C (skipped out the function and header files): 
```c
    af_index_t* indexArr =  (af_index_t*)(malloc(4 * sizeof(af_index_t)));
    af_err err = af_create_indexers(&indexArr);
    printf("Index_t size:  %lu \n", sizeof(af_index_t));
    printf("Error code:  %d \n", err);

    //Create array
    af_array a;
    unsigned ndims = 2;
    dim_t dim[] = {10,10};
    af_randu(&a, ndims, dim, f32);
    af_print_array(a);

    //Create index array
    af_array idx;
    unsigned n = 1;
    dim_t d[] = {5};
    af_range(&idx, n, d, 0, s64);
    af_print_array(idx);

    //create array indexer
    af_err err2 = af_set_array_indexer(indexArr, idx, 1);
    printf("Error from set array indexer: %d \n", err2);

    //index with indexArr
    af_array out;
    af_index_gen(&out, a, 1, indexArr);  //segfaults
    af_print_array(out);
    printf("Yay this works! \n");

    //create another array indexer 
    af_err err3 = af_set_array_indexer(indexArr, idx, 0);
    printf("Error from second array indexer: %d \n", err3);

    //index with indexArr
    af_array out2;
    af_index_gen(&out2, a, 2, indexArr);
    af_print_array(out2);
```
When I set talk about indexing in the second dimension, my index function segfaults at that particular line with the comment. This is because the `create_indexers` creates garbage objects by default. 

After a discussion with @9prady9 and @pavanky , it seems that by default, `af_create_indexers` should return `span,span,span,span` so users can set any dimension directly.

**Comment by pavanky** [comment]
@9prady9 can you write up an example as well in the documentation for this as well ?

**Comment by 9prady9** [comment]
Resolved by #1426
